community/index/src/main/java/org/neo4j/index/internal/gbptree/package-info.java

```diff
@@ -18,13 +18,135 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 /**
- * B+tree implementation with arbitrary key/value. Index implementation is
+ * <h1>The GBPTree</h1>
+ * B+tree implementation with arbitrary key:value entries. Index implementation is
  * {@link org.neo4j.index.internal.gbptree.GBPTree}, which works on a {@link org.neo4j.io.pagecache.PageCache}.
  * Implementation supports single writer together with concurrent lock-free and garbage-free readers.
  * <p>
- * To create an index with a custom layout (type of key/value), implement a custom
- * {@link org.neo4j.index.internal.gbptree.Layout}.
+ * To create an index with a custom layout (type of key:value), implement a custom {@link org.neo4j.index.internal.gbptree.Layout}.
  * <p>
  * See https://en.wikipedia.org/wiki/B%2B_tree
+ * <h1>Entries move from left to right</h1>
+ * When we make structural changes to the tree we are only allowed to move entries to the right, never to the left.
+ * {@link org.neo4j.index.internal.gbptree.SeekCursor Readers} rely on this assumption being true to read correct
+ * entries without missing anything. This is also way backwards seek needs to be done with extra care. Moving entries
```

Comment:
Typo: `...This is also way backwards seek needs...` --> `...This is also why backwards seek needs...`